src/lib.rs

```diff
@@ -69,38 +91,114 @@ fn dispute(params: Params, state: FullySignedState) -> Option<Error> {
 /// withdrawable.
 fn conclude(params: Params, state: FullySignedState) -> Option<Error> {
 	STATE
-		.with(|s| s.borrow_mut().conclude(params, state, blocktime()))
+		.write()
+		.unwrap()
+		.conclude(params, state, blocktime())
 		.err()
 }
 
 #[ic_cdk_macros::update]
 /// Withdraws the specified participant's funds from a settled channel.
-fn withdraw(request: WithdrawalRequest, auth: L2Signature) -> (Option<Amount>, Option<Error>) {
-	let result = STATE.with(|s| s.borrow_mut().withdraw(request, auth, blocktime()));
+async fn withdraw(
+	request: WithdrawalRequest,
+	auth: L2Signature,
+) -> (Option<icp::BlockHeight>, Option<Error>) {
+	let result = withdraw_impl(request, auth).await;
```

Comment:
Why separate `withdraw_impl` from `withdraw`?

Reply:
because of the `?` operator which cannot be used in non-`Result` functions.